core/src/main/scala/io/odin/formatter/Formatter.scala

```diff
@@ -134,7 +139,11 @@ object Formatter {
         builder.append(t.getLocalizedMessage)
       }
       builder.append(System.lineSeparator())
-      writeStackTrace(builder, depth.fold(t.getStackTrace)(t.getStackTrace.take), indent)
+
+      val filteredStackTrace = filter.fold(t.getStackTrace)(set =>
+        t.getStackTrace.filterNot(row => set.contains(row.getClassName.stripSuffix("$")))
```

Comment:
Would it be faster to move this check into `writeStackTrace` loop and save few allocations?

Reply:
then we'll need to move the `t.getStackTrace.take` part into `writeStackTrace` loop too, because we could `take(5)` and then filter them out, having empty list of rows.

Reply:
yeah, it's possible to just count amount of actual lines printed and stop whenever `depth` is achieved

no worries, it's a matter of optimization in separate PR then

Reply:
Can I make this PR for optimization?

Reply:
@lightning95 sorry haven't noticed it earlier

yes, feel free to